### PR TITLE
add e2e tests for service accounts

### LINKED ISSUE #3455: Create e2e test for kubermatic API
**User Story**
Create unit tests to check Kubermatic API endpoints for Service Account
**Acceptance criteria**
 - fix swagger spec and generate API client
- use API client to get access to Service Account endpoints
- create test scenarios for Service Accounts

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: e2e tests for API. Tests scenarios for service accounts

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3455

```release-note
NONE
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
@zreigz we could generate API client from our swagger spec (https://github.com/kubermatic/kubermatic/blob/master/api/hack/update-swagger.sh)

the net profit is that we get a strongly typed client and we don't have to build it manually (essentially everything in `api/test/e2e/helper.go`), wdyt ?

I think it should be really ease to generate a client from the spec, you can have a look at https://github.com/go-swagger/go-swagger

**Comment by zreigz** [comment]
@p0lyn0mial I will try this API client, looks interesting

**Comment by p0lyn0mial** [comment]
> pull-kubermatic-api-e2e — Job succeeded.

:)

**Comment by p0lyn0mial** [comment]
> PASS: TestTokenAccessForProject/test_project_access_when_token_has_editor_rights (20.99s)
helper.go:100: project created: [POST /api/v1/projects][201] createProjectCreated &{CreationTimestamp:2019-05-16T03:34:44.000Z DeletionTimestamp:0001-01-01T00:00:00.000Z ID:d8btw2xm64 Name:7vb7jkzf8l Owners:[0xc0004acb00] Status:Inactive}
helper.go:189: service account created: [POST /api/v1/projects/{project_id}/serviceaccounts][201] addServiceAccountToProjectCreated &{CreationTimestamp:2019-05-16T03:34:46.000Z DeletionTimestamp:0001-01-01T00:00:00.000Z Group:editors-d8btw2xm64 ID:mtnmxg9s4v Name:cwnjnzhd82 Status:Inactive}
helper.go:260: service account token created: [POST /api/v1/projects/{project_id}/serviceaccounts/{serviceaccount_id}/tokens][201] addTokenToServiceAccountCreated &{CreationTimestamp:2019-05-16T03:34:46.000Z DeletionTimestamp:0001-01-01T00:00:00.000Z Expiry:2022-05-16T03:34:46.000Z ID:6th8qhs4s9 Name:zdpv2g5z27 Token:eyJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6InNlcnZpY2VhY2NvdW50LW10bm14ZzlzNHZAY2kua3ViZXJtYXRpYy5pbyIsImV4cCI6MTY1MjY3MjA4NiwiaWF0IjoxNTU3OTc3Njg2LCJuYmYiOjE1NTc5Nzc2ODYsInByb2plY3RfaWQiOiJkOGJ0dzJ4bTY0IiwidG9rZW5faWQiOiI2dGg4cWhzNHM5In0.qT6K6PlowtKg4w4b9qSqj7UMQp_8o2-cOw-_3gBIkJI}
helper.go:100: project created: [POST /api/v1/projects][201] createProjectCreated &{CreationTimestamp:2019-05-16T03:34:46.000Z DeletionTimestamp:0001-01-01T00:00:00.000Z ID:gx5hb8k76g Name:gxd8gxk2jt Owners:[0xc0003b5880] Status:Inactive}


Would be nice if you could make the output of a running test more readable. Perhaps we should only log the errors and annotate them a bit.

**Comment by zreigz** [comment]
@p0lyn0mial PTAL

File: api/pkg/handler/routes_v1.go [hunk 1]
```diff
@@ -1,3 +1,19 @@
+// Kubermatic API.
```
Comment (floreks) [inline api/pkg/handler/routes_v1.go:1]: nit: Is this supposed to be here?

Comment (zreigz) [reply api/pkg/handler/routes_v1.go:1]: yes, without this swagger remove the title from doc

**Pull request merged**